src/QsCompiler/Compiler/RewriteSteps/IntrinsicMapping.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Quantum.QsCompiler.SyntaxTree;
+using Microsoft.Quantum.QsCompiler.Transformations.IntrinsicMappingTransformation;
+
+
+namespace Microsoft.Quantum.QsCompiler.BuiltInRewriteSteps
+{
+    internal class IntrinsicMapping : IRewriteStep
+    {
+        public string Name { get; }
+        public int Priority { get; }
+        public string OutputFolder { get; set; }
+
+        public bool ImplementsTransformation { get; }
+        public bool ImplementsPreconditionVerification { get; }
+        public bool ImplementsPostconditionVerification { get; }
```

Comment:
Given that the mapping itself blindly replaces declarations with the same full name, a postcondition verification seems useful (possibly in a separate PR). And I am tempted to suggest to keep an internal state with the original signatures that got replaced in here, such that the postcondition verification doesn't need to walk the whole tree but can just compare those with the replaced ones.

Reply:
I'm not clear what you are suggesting we check for in post-verification. Could you please clarify?

Reply:
Say there was by accident a callable with the same full name in the environment and in the compilation but they had different signatures, then the tree after pulling in the environment specific declarations is broken. It would be good to have the option to verify that. 
Concretely, that means that you could have an internal state in the rewrite step that keeps the signatures of all replaced callables (e.g. a dictionary of QsQualifiedName to ResolvedSignature) that is filled during the Transformation, and then a postcondition verification that given the built compilation, checks that all callables in that declaration are either not present in the dictionary or have the same signature as those in the dictionary.